www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php

```diff
@@ -108,102 +129,158 @@
-    . $stringOk . " " . $stringMeanTime . ";" . $stringOk . " " . $stringAlert . ";"
-    . $stringWarning . " " . $stringMeanTime . ";" . $stringWarning . " " . $stringAlert . ";"
-    . $stringUnknown . " " . $stringMeanTime . ";" . $stringUnknown . " " . $stringAlert . ";"
-    . $stringCritical . " " . $stringMeanTime . ";" . $stringCritical . " " . $stringAlert . ";"
+// Services group stats evolution
+echo _("Day") . ";"
+    . _("Duration") . ";"
+    . $stringOk . " " . $stringMeanTime . ";"
+    . $stringOk . " " . $stringAlert . ";"
+    . $stringWarning . " " . $stringMeanTime . ";"
+    . $stringWarning . " " . $stringAlert . ";"
+    . $stringUnknown . " " . $stringMeanTime . ";"
+    . $stringUnknown . " " . $stringAlert . ";"
+    . $stringCritical . " " . $stringMeanTime . ";"
+    . $stringCritical . " " . $stringAlert . ";"
     . _("Day") . "\n";
 
+$dbResult = $pearDB->query(
+    "SELECT `service_service_id` FROM `servicegroup_relation` " .
+    "WHERE `servicegroup_sg_id` = '" . $servicegroupId . "'"
+);
+
 $str = "";
-$request = "SELECT `service_service_id` FROM `servicegroup_relation` WHERE `servicegroup_sg_id` = '".$id."'";
-$DBRESULT = $pearDB->query($request);
-while ($sg = $DBRESULT->fetchRow()) {
+while ($sg = $dbResult->fetch()) {
     if ($str != "") {
         $str .= ", ";
     }
     $str .= "'" . $sg["service_service_id"] . "'";
 }
-$DBRESULT->closeCursor();
+$dbResult->closeCursor();
 if ($str == "") {
     $str = "''";
 }
 unset($sg);
-unset($DBRESULT);
-
-$request =  "SELECT `date_start`, `date_end`, sum(`OKnbEvent`) as OKnbEvent, sum(`CRITICALnbEvent`) as CRITICALnbEvent,"
-            . " sum(`WARNINGnbEvent`) as WARNINGnbEvent, sum(`UNKNOWNnbEvent`) as UNKNOWNnbEvent, "
-            . "avg( `OKTimeScheduled` ) as OKTimeScheduled, "
-            . "avg( `WARNINGTimeScheduled` ) as WARNINGTimeScheduled, "
-            . "avg( `UNKNOWNTimeScheduled` ) as UNKNOWNTimeScheduled, "
-            . "avg( `CRITICALTimeScheduled` ) as CRITICALTimeScheduled "
-            . "FROM `log_archive_service` WHERE `service_id` IN (".$str.") "
-            . "AND `date_start` >= '".$start_date."' "
-            . "AND `date_end` <= '".$end_date."' "
-            . "GROUP BY `date_end`, `date_start` order by `date_start` desc";
-$res = $pearDBO->query($request);
+unset($dbResult);
+
+$res = $pearDBO->query(
+    "SELECT `date_start`, `date_end`, sum(`OKnbEvent`) as OKnbEvent, "
+    . "sum(`CRITICALnbEvent`) as CRITICALnbEvent, "
+    . "sum(`WARNINGnbEvent`) as WARNINGnbEvent, "
+    . "sum(`UNKNOWNnbEvent`) as UNKNOWNnbEvent, "
+    . "avg( `OKTimeScheduled` ) as OKTimeScheduled, "
+    . "avg( `WARNINGTimeScheduled` ) as WARNINGTimeScheduled, "
+    . "avg( `UNKNOWNTimeScheduled` ) as UNKNOWNTimeScheduled, "
+    . "avg( `CRITICALTimeScheduled` ) as CRITICALTimeScheduled "
+    . "FROM `log_archive_service` WHERE `service_id` IN (" . $str . ") "
+    . "AND `date_start` >= '" . $start_date . "' "
```

Comment:
In this case, using a prepare statement is hightly recommended

Reply:
Refacto the whole code is out of my scope...  ;)